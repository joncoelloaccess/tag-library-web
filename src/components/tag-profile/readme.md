# tag-profile

















## Demos

















#### Basic profile

















```html








<tag-profile name="Kevin Anderson"></tag-profile>








```

















#### Optional Gender

















```html








<tag-profile








  name="Kevin Anderson"








  gender="Male"








  dob="08/Aug/1982"








  nationality="British"








  ></tag-profile>








```

















#### With edit profile event listener

















```html








<tag-profile








  name="Kevin Anderson"








  gender="Male"








  dob="08/Aug/1982"








  nationality="British"








  src="http://ux.accessacloud.com/StyleGuide/Live/Content/img/rachel.jpg"








  ></tag-profile>

















<script>








  window.addEventListener("editProfileImage", function () {








    console.log('yo yo yo');








  });








</script>








```

















<!-- Auto Generated Below -->


## Properties

#### dob

string


#### gender

string


#### name

string

Name of user profiled


#### nationality

string


#### src

string

Src of profile image


## Attributes

#### dob

string


#### gender

string


#### name

string

Name of user profiled


#### nationality

string


#### src

string

Src of profile image


## Events

#### editProfileImage



----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
