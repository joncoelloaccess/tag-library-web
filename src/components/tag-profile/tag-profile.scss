@import "../../shared/style-atoms";

:host {
  text-align: center;
  hr {
    border-top: 1px solid $borders;
  }
  .prof {
    &__img {
      display: inline-block;
      height: 230px;
      width: 230px;
      background-color: $viridian-green;
      border-radius: 115px;
      position: relative;
    }
    &__edit {
      position: absolute;
      height: 44px;
      width: 44px;
      border-radius: 23px;
      background-color: $viridian-green;
      right: 0;
      bottom: 30px;
      cursor: pointer;
    }
  }
}
