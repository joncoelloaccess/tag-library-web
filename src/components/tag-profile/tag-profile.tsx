import { Component, Event, EventEmitter, Prop } from "@stencil/core";

@Component({
  tag: "tag-profile",
  styleUrl: "tag-profile.scss",
  shadow: true
})
export class TagProfile {
  /** Name of user profiled */
  @Prop() name: string;
  /** Src of profile image */
  @Prop() src: string;
  @Prop() gender: string;
  @Prop() dob: string;
  @Prop() nationality: string;
  @Event() editProfileImage: EventEmitter;
  onEditProfileImage() {
    this.editProfileImage.emit();
  }
  render() {
    const { dob, gender, name, nationality, src } = this;

    return (
      <div>
        <div
          class="prof__img"
          style={{
            backgroundSize: "cover",
            backgroundImage: `url(${src})`
          }}
        >
          <div class="prof__edit" onClick={() => this.onEditProfileImage()}>
            <tag-icon-set
              icon="Edit"
              accent="Selected"
              style={{
                width: "24px",
                height: "24px",
                padding: "10px"
              }}
            />
          </div>
        </div>
        <tag-text type="h2">{name}</tag-text>
        <hr />
        {gender ? <tag-text type="h3">Gender - {gender}</tag-text> : null}
        {dob ? <tag-text type="h3">Date of birth - {dob}</tag-text> : null}
        {nationality ? (
          <tag-text type="h3">Nationality - {nationality}</tag-text>
        ) : null}
      </div>
    );
  }
}
