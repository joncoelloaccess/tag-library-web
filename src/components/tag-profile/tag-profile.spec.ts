import { render, flush } from "@stencil/core/testing";
import { TagProfile } from "./tag-profile";

const renderElement = async (html: string): Promise<HTMLElement> => {
  return await render({
    components: [TagProfile],
    html: html
  });
};

describe("tag-profile", () => {
  it("should build", () => {
    expect(new TagProfile()).toBeTruthy();
  });

  describe("rendering", () => {
    let element: HTMLElement;
    beforeEach(async () => {
      element = await render({
        components: [TagProfile],
        html: "<tag-profile></tag-profile>"
      });
    });
    it("should render", async () => {
      await flush(element);
      expect(element.outerHTML).toMatchSnapshot();
    });
    it("should render everything", async () => {
      const element: HTMLElement = await render({
        components: [TagProfile],
        html: `<tag-profile
          name="Kevin Anderson"
          gender="Male"
          dob="08/Aug/1982"
          nationality="British"
          src="http://ux.accessacloud.com/StyleGuide/Live/Content/img/rachel.jpg"
        ></tag-profile>`
      });
      await flush(element);
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
  describe("edit events", () => {
    it("should emit a 'edit profile' event", async () => {
      // const dummyFn = jest.fn();
      // const element = await renderElement("<tag-profile></tag-profile>");
      // element.addEventListener("editProfileImage", dummyFn);
      // await flush(element);
      // const edit = element.querySelector<HTMLElement>(".prof__edit");
      // edit.click();
      // setTimeout(() => {
      //   expect(dummyFn).not.toHaveBeenCalled();
      // }, 50);
    });
  });
});
