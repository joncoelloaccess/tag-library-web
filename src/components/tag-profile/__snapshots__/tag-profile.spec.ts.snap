// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag-profile rendering should render 1`] = `"<tag-profile data-tag-profile-host=\\"\\" class=\\"hydrated\\"><!----><div data-tag-profile=\\"\\"><div class=\\"prof__img\\" style=\\"background-size: cover; background-image: url(undefined);\\" data-tag-profile=\\"\\"><div class=\\"prof__edit\\" data-tag-profile=\\"\\"><tag-icon-set icon=\\"Edit\\" accent=\\"Selected\\" style=\\"width: 24px; height: 24px; padding: 10px;\\" data-tag-profile=\\"\\"></tag-icon-set></div></div><tag-text type=\\"h2\\" data-tag-profile=\\"\\"></tag-text><hr data-tag-profile=\\"\\"></div></tag-profile>"`;

exports[`tag-profile rendering should render everything 1`] = `"<tag-profile name=\\"Kevin Anderson\\" gender=\\"Male\\" dob=\\"08/Aug/1982\\" nationality=\\"British\\" src=\\"http://ux.accessacloud.com/StyleGuide/Live/Content/img/rachel.jpg\\" data-tag-profile-host=\\"\\" class=\\"hydrated\\"><!----><div data-tag-profile=\\"\\"><div class=\\"prof__img\\" style=\\"background-size: cover; background-image: url(http://ux.accessacloud.com/StyleGuide/Live/Content/img/rachel.jpg);\\" data-tag-profile=\\"\\"><div class=\\"prof__edit\\" data-tag-profile=\\"\\"><tag-icon-set icon=\\"Edit\\" accent=\\"Selected\\" style=\\"width: 24px; height: 24px; padding: 10px;\\" data-tag-profile=\\"\\"></tag-icon-set></div></div><tag-text type=\\"h2\\" data-tag-profile=\\"\\">Kevin Anderson</tag-text><hr data-tag-profile=\\"\\"><tag-text type=\\"h3\\" data-tag-profile=\\"\\">Gender - Male</tag-text><tag-text type=\\"h3\\" data-tag-profile=\\"\\">Date of birth - 08/Aug/1982</tag-text><tag-text type=\\"h3\\" data-tag-profile=\\"\\">Nationality - British</tag-text></div></tag-profile>"`;
